include/counter.h

```diff
@@ -29,45 +29,35 @@
 extern "C" {
 #endif
 
-struct counter_alarm_cfg;
-
 /** @brief Alarm callback
  *
- * @param dev     Pointer to the device structure for the driver instance.
- * @param cfg     Original structure owning this handler.
- * @param ticks   Counter value that triggered the callback.
+ * @param user_data User data.
+ * @param chan_id   Channel ID.
+ * @param ticks     Counter value that triggered the callback.
  */
-typedef void (*counter_alarm_callback_t)(struct device *dev,
-					 const struct counter_alarm_cfg *cfg,
+typedef void (*counter_alarm_callback_t)(void *user_data, u8_t chan_id,
 					 u32_t ticks);
 
 /** @brief Alarm callback structure.
  *
- * Used to set an alarm.
- * Beware such structure should not be allocated on stack.
+ * @param handler Handler called on alarm (cannot be NULL).
+ * @param ticks Ticks that triggers the alarm. In case of absolute flag is set,
+ *		maximal value that can be set equals wrap value
+ *		(counter_get_wrap). Otherwise counter_get_max_relative_alarm()
+ *		returns maximal value that can be set. If counter is clock
+ *		driven then ticks can be converted to microseconds (see @ref
+ *		counter_ticks_to_us). Alternatively, counter implementation may
+ *		count asynchronous events.
+ * @param user_data User data returned in callback.
+ * @param absolute Ticks relation to counter value. If true ticks are treated as
+ *		absolute value, else it is relative to the counter reading
+ *		performed during the call.
  */
 struct counter_alarm_cfg {
-	counter_alarm_callback_t handler; /*!< handler called on alarm
-					   *   (cannot be NULL).
-					   */
-
-	u32_t ticks; /*!< In case of absolute flag is set, maximal value that
-		      *   can be set equals wrap value (counter_get_wrap).
-		      *   Otherwise counter_get_max_relative_alarm() returns
-		      *   maximal value that can be set. If counter is clock
-		      *   driven then ticks can be converted to microseconds
-		      *   (see @ref counter_ticks_to_us). Alternatively,
-		      *   counter implementation may count asynchronous events.
-		      */
-
-	u8_t channel_id; /*!< Channel ID. Number of channels is driver
-			  * dependent.
-			  */
-
-	bool absolute; /*!< If true ticks are treated as absolute value, else
-			*   it is relative to the counter reading performed
-			*   during the call.
-			*/
+	counter_alarm_callback_t handler;
```

Comment:
Let's rename this field to `callback`. That's also how other APIs e.g. watchdog.h name this field.

Reply:
renamed.